src/test/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentStatusServiceSpec.scala

```diff
@@ -350,13 +380,261 @@ class ConsignmentStatusServiceSpec extends AnyFlatSpec with MockitoSugar with Re
+    )
+    when(
+      consignmentStatusRepositoryMock.updateConsignmentStatus(
+        any[UUID],
+        any[String],
+        any[String],
+        any[Timestamp]
+      )
+    ).thenReturn(consignmentStatusMockRepoResponse)
+
+    when(
+      fileStatusRepositoryMock.getFileStatus(
+        consignmentIdCaptor.capture(),
+        statusTypeCaptor.capture(),
+        any[Option[Set[UUID]]]
+      )
+    ).thenReturn(fileStatusMockRepoResponse)
+
+    val updateConsignmentStatusInput =
+      ConsignmentStatusInput(expectedConsignmentId, expectedStatusType, None)
+
+    consignmentService.updateConsignmentStatus(updateConsignmentStatusInput).futureValue
+
+    consignmentIdCaptor.getValue should equal(expectedConsignmentId)
+    statusTypeCaptor.getValue should equal(expectedStatusType)
+  }
+
+  forAll(statusTypes.filter(_ != "Upload")) { nonUploadStatusType =>
+    "updateConsignmentStatus" should s"not call fileStatus repo if a non-'Upload' statusType of ${nonUploadStatusType} has been passed in" in {
+      val fixedUUIDSource = new FixedUUIDSource()
+      val expectedConsignmentId = fixedUUIDSource.uuid
+      val expectedStatusValue = "Completed"
+      val modifiedTime = Timestamp.from(FixedTimeSource.now)
+      val consignmentIdCaptor: ArgumentCaptor[UUID] = ArgumentCaptor.forClass(classOf[UUID])
+      val statusTypeCaptor: ArgumentCaptor[String] = ArgumentCaptor.forClass(classOf[String])
+      val statusValueCaptor: ArgumentCaptor[String] = ArgumentCaptor.forClass(classOf[String])
+      val modifiedTimeCaptor: ArgumentCaptor[Timestamp] = ArgumentCaptor.forClass(classOf[Timestamp])
+
+      val consignmentStatusMockRepoResponse: Future[Int] = Future.successful(1)
+
+      when(
+        consignmentStatusRepositoryMock.updateConsignmentStatus(
+        consignmentIdCaptor.capture(),
+        statusTypeCaptor.capture(),
+        statusValueCaptor.capture(),
+        modifiedTimeCaptor.capture()
+        )
+      ).thenReturn(consignmentStatusMockRepoResponse)
+
+      val updateConsignmentStatusInput =
+        ConsignmentStatusInput(expectedConsignmentId, nonUploadStatusType, Some(expectedStatusValue))
+
+      val response: Int = consignmentService.updateConsignmentStatus(updateConsignmentStatusInput).futureValue
+
+      verify(fileStatusRepositoryMock, times(0)).getFileStatus(
+        any[UUID],
+        any[String],
+        any[Option[Set[UUID]]]
+      )
+      response should be(1)
+      consignmentIdCaptor.getValue should equal(expectedConsignmentId)
+      statusTypeCaptor.getValue should equal(nonUploadStatusType)
+      statusValueCaptor.getValue should equal(expectedStatusValue)
+      modifiedTimeCaptor.getValue should equal(modifiedTime)
+    }
+  }
+
+  forAll(statusTypes.filter(_ != "Upload")) { nonUploadStatusType =>
+    "updateConsignmentStatus" should s"throw an InputDataException when only a consignmentId and a non-'Upload' statusType of ${nonUploadStatusType} has been passed in" in {
```

Comment:
Maybe re-word test title slightly to: `... when a consignmentId and a non-'Upload' ...`